pulp_ansible/app/tasks/utils.py

```diff
@@ -0,0 +1,52 @@
+import json
+
+from urllib.parse import parse_qs, urlencode, urlparse, urlunparse
+from aiohttp import ClientSession
+
+
+async def get_json_from_url(url):
+    """Asynchronously get the json from request."""
+    async with ClientSession() as session:
```

Comment:
This creates a new session each time this function is called. I think we need to bring in Pulp's `DownloadFactory` to generate these downloaders. aiohttp likes to share sessions.

Also note that by creating these downloaders directly (without the Remote's settings like proxy settings) if the Pulp system is behind a proxy this connection won't work.

Here are the docs explaining how to make downloaders from your Remote.  https://docs.pulpproject.org/en/pulpcore-plugin/nightly/api-reference/download.html#configuring-from-a-remote